compiler/il/OMROpcodes.hpp

```diff
@@ -0,0 +1,795 @@
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_OPCODEMACROS_INCL
+#define OMR_OPCODEMACROS_INCL
+
+/**
+ *
+ * Opcodes consist of (in order):
+ * - opcode
+ * - name
+ * - properties1
+ * - properties2
+ * - properties3
+ * - properties4
+ * - dataType
+ * - typeProperties
+ * - childProperties
+ * - swapChildrenOpcode
+ * - reverseBranchOpcode
+ * - booleanCompareOpcode
+ * - ifCompareOpcode
+ *
+ */
+
+#define FOR_EACH_OPCODE(MACRO) \
+   MACRO(\
+      /* opcode */ TR::BadILOp,\
+      "BadILOp",\
+      0,\
+      0,\
+      0,\
+      0,\
+      TR::NoType,\
+      0,\
+      ILChildProp::NoChildren,\
+      TR::BadILOp,\
+      TR::BadILOp,\
+      TR::BadILOp,\
+      TR::BadILOp) \
+   MACRO(\
+      TR::aconst,\
+      "aconst",ILProp1::LoadConst,ILProp2::ValueNumberShare,ILProp3::LikeUse,0,TR::Address,ILTypeProp::Reference,ILChildProp::NoChildren,TR::BadILOp,TR::BadILOp,TR::BadILOp,TR::BadILOp) \
+   MACRO(TR::iconst,"iconst",ILProp1::LoadConst,ILProp2::ValueNumberShare,ILProp3::LikeUse,0,TR::Int32,ILTypeProp::Size_4 | ILTypeProp::Integer,ILChildProp::NoChildren,TR::BadILOp,TR::BadILOp,TR::BadILOp,TR::BadILOp) \
+   MACRO(TR::lconst,"lconst",ILProp1::LoadConst,ILProp2::ValueNumberShare,ILProp3::LikeUse,0,TR::Int64,ILTypeProp::Size_8 | ILTypeProp::Integer,ILChildProp::NoChildren,TR::BadILOp,TR::BadILOp,TR::BadILOp,TR::BadILOp) \
+   MACRO(TR::fconst,"fconst",ILProp1::LoadConst,ILProp2::ValueNumberShare,ILProp3::LikeUse,0,TR::Float,ILTypeProp::Size_4 | ILTypeProp::Floating_Point,ILChildProp::NoChildren,TR::BadILOp,TR::BadILOp,TR::BadILOp,TR::BadILOp) \
+   MACRO(TR::dconst,"dconst",ILProp1::LoadConst,ILProp2::ValueNumberShare,ILProp3::LikeUse,0,TR::Double,ILTypeProp::Size_8 | ILTypeProp::Floating_Point,ILChildProp::NoChildren,TR::BadILOp,TR::BadILOp,TR::BadILOp,TR::BadILOp) \
+   MACRO(TR::bconst,"bconst",ILProp1::LoadConst,ILProp2::ValueNumberShare,ILProp3::LikeUse,0,TR::Int8,ILTypeProp::Size_1 | ILTypeProp::Integer,ILChildProp::NoChildren,TR::BadILOp,TR::BadILOp,TR::BadILOp,TR::BadILOp) \
```

Comment:
Before you get too far into this I'd point out that at least to me the readability of these definitions has regressed. However these would just be formatting changes we could decide on later.

Reply:
Thanks for the feedback! Definitely, there is a need for more readability there. I have the following ideas:

1. Since there are more properties to add, I want to turn single-lined macro arguments to multiple lines. Otherwise, the line length may easily exceed 100 chars and it is going to look stuffed and quite messy.

2. If we follow (1), we can actually append a comment to each line so developers reading that `OMROpcodes.hpp` know what each argument means. This could be automatically generated (as I generated `OMROpcodes.hpp` using scripts).

Reply:
Sounds reasonable. If we can maintain a similar look to what we have in OMRILOpCodeProperties.hpp I think most people will be happy.